lazybar-core: mpd: Code cleanup

cecilialau6776 opened the pull request:
- Created type alias for complex type (`pub type IndexCache = Vec<ButtonIndex>;`
  - Turned the tuple into a `ButtonIndex`
- General code cleanup - moved repeated code into a function/out of `match`es

Commit:
lazybar-core: mpd: create type alias for complex type

Qelxiros reviewed (changes requested):
I'd like to see `IndexCache` as a public typedef (e.g. in `lib.rs` with the others). The mpd panel is the first to support dynamic, user-defined clickable areas, but I don't expect it to be the last. I haven't found a way to implement it without an index cache, so exposing that seems good. Plus, we get more documentation that way :)

Commit:
lazybar-core: mpd: clean up code

cecilialau6776 commented:
> I'd like to see `IndexCache` as a public typedef (e.g. in `lib.rs` with the others). The mpd panel is the first to support dynamic, user-defined clickable areas, but I don't expect it to be the last. I haven't found a way to implement it without an index cache, so exposing that seems good. Plus, we get more documentation that way :)

Implemented in [42d7bbe](https://github.com/Qelxiros/lazybar/pull/11/commits/42d7bbe4329fab7f2a0c6c65ec6ff1390ba519c1)

Also updated the PR message to reflect new changes

Qelxiros committed:
lazybar-core: fix spelling
